src/main/java/ru/hogwarts/school/service/StudentService.java

```diff
@@ -1,38 +1,90 @@
 package ru.hogwarts.school.service;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import ru.hogwarts.school.model.Avatar;
+import ru.hogwarts.school.model.Faculty;
 import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repository.AvatarRepository;
+import ru.hogwarts.school.repository.StudentRepository;
 
-import java.util.ArrayList;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Collection;
-import java.util.HashMap;
+import java.util.stream.Collectors;
+
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+
 @Service
 public class StudentService {
-    HashMap<Long, Student> students = new HashMap<>();
-    long lastId = 0;
+    @Value("${avatars.dir.path}")
+    private String avatarsDir;
+    private final StudentRepository studentRepository;
+    private final AvatarRepository avatarRepository;
 
-    public Student createStudent(Student student){
-        student.setId(++lastId);
-        return students.put(lastId, student);
+    public StudentService(StudentRepository studentRepository, AvatarRepository avatarRepository) {
+        this.studentRepository = studentRepository;
+        this.avatarRepository = avatarRepository;
     }
-    public Student findStudent(long id){
-        return students.get(id);
+
+
+    public Student createStudent(Student student) {
+        return studentRepository.save(student);
     }
-    public Student editStudent(Student student){
-        return students.put(student.getId(), student);
+
+    public Student findStudent(long id) {
+        return studentRepository.findById(id).get();
     }
-    public Student deleteStudent(long id){
-        return students.remove(id);
+
+    public Student editStudent(Student student) {
+        return studentRepository.save(student);
+    }
+
+    public void deleteStudent(long id) {
+        studentRepository.deleteById(id);
     }
 
-    public Collection<Student> findAge(int age){
-        ArrayList<Student>result = new ArrayList<>();
-        for (Student student: students.values()){
-            if (student.getAge() == age){
-                result.add(student);
-            }
+    public Collection<Student> getAllByAge(int age) {
+        return studentRepository.findAll()
+                .stream()
+                .filter(student -> student.getAge() == age)
+                .collect(Collectors.toList());
+    }
+
+    public Collection<Student> findByAgeBetween(int min, int max) {
+        return studentRepository.findByAgeBetween(min, max);
+    }
+
+    public Avatar findAvatar(Long studentId) {
+        return AvatarRepository.findByStudentId(studentId).orElseThrow();
```

Comment:
```suggestion
        return avatarRepository.findByStudentId(studentId).orElseThrow();
```
avatarRepository с маленькой буквы